docs/ds/dividing.md

```diff
@@ -1,45 +1,45 @@
 author: Xarfa
 
-划分树是一种来解决区间第 $K$ 大的一种数据结构，其常数、理解难度都要比主席树低很多。同时，划分树紧贴“第 $K$ 大”，所以是一种基于排序的一种数据结构。
+The dividing tree is a data structure to solve the largest $K$ interval, and its time complexity constant and the difficulty of understanding are much lower than the [persistent segment tree](../persistent-seg/#_1). At the same time, the dividing tree closely adheres to the "largest $K$ ", so it is a sort-based data structure.
 
- **建议先学完 [主席树](../persistent-seg/#_1) 再看划分树哦** 
+ **It is recommended to finish learning [Persistent Segment Tree](../persistent-seg/#_1) before learning dividing tree**
 
-## 建树
+## Tree construction
 
-划分树的建树比较简单，但是相对于其他树来说比较复杂。![](./images/dividing1.png)
+The construction of the dividing tree is relatively simple, but it is more complicated than other trees. ![](./images/dividing1-en.png)
 
-如图，每一层都有一个看似无序的数组。其实，每一个被红色标记的数字都是 **要分配到左儿子的** 。而分配的规则是什么？就是与 **这一层的中位数** 做比较， $\leq$ 左边，否则右边。但是这里要注意一下：并不是严格的 $\leq$  **左边，否则右边** 。因为中位数可能有相同，而且与 $N$ 的奇偶有一定关系。下面的代码展示会有一个巧妙的运用，大家可以参照代码。
+As shown in the figure, each layer has a seemingly unordered array. In fact, every number marked in red is **to be assigned to the left child**. And the rules of distribution? It is compared with the median in this level, $\leq$ is on the left, otherwise it is on the right. But here we should pay attention: it is not strictly $\leq$ **left, otherwise right**. Because the median may be the same, and it has a certain relationship with the parity of $N$ . The following code would demonstrate a clever usage for your reference.
+ 
+We cannot sort each layer every time, because the theoretical complexity would not pass, not to mention the constants. To find the median, one pass sorting is enough. Why? For example, if we find the median of $l,r$ , it is actually $num[mid]$ after sorting.
```

Comment:
```suggestion
We cannot sort each layer every time, because the theoretical complexity would not allow, not to mention the constants. To find the median, one pass sorting is enough. Why? For example, if we find the median of $l,r$ , it is actually `num[mid]` after sorting.
```